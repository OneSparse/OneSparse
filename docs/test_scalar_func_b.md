# Scalar 'bool' Functions

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
``` postgres-console
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
```
Test construction of min, zero and max values:
``` postgres-console
select '0'::bool::scalar;
 scalar 
--------
 b:f
(1 row)

select '0'::bool::scalar;
 scalar 
--------
 b:f
(1 row)

select '1'::bool::scalar;
 scalar 
--------
 b:t
(1 row)

```
Test setting a scalar value from max to 2
``` postgres-console
select set('1'::bool::scalar, 2);
 set 
-----
 b:t
(1 row)

```
Test various casting functions used by the CREATE CAST machinery:
``` postgres-console
select scalar_bool((0)::bool);
 scalar_bool 
-------------
 b:f
(1 row)

select scalar_bool((0)::bool);
 scalar_bool 
-------------
 b:f
(1 row)

select scalar_bool((1)::bool);
 scalar_bool 
-------------
 b:t
(1 row)

select bool_scalar((0)::bool::scalar);
 bool_scalar 
-------------
 f
(1 row)

select bool_scalar((0)::bool::scalar);
 bool_scalar 
-------------
 f
(1 row)

select bool_scalar((1)::bool::scalar);
 bool_scalar 
-------------
 t
(1 row)

```