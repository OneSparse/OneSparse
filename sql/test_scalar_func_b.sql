-- # Scalar `bool` Functions
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

-- Test construction of min, zero and max values:
select '0'::bool::scalar;
select '0'::bool::scalar;
select '1'::bool::scalar;

-- Test setting a scalar value from max to 2
select set('1'::bool::scalar, 2);

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_bool((0)::bool);
select scalar_bool((0)::bool);
select scalar_bool((1)::bool);
select bool_scalar((0)::bool::scalar);
select bool_scalar((0)::bool::scalar);
select bool_scalar((1)::bool::scalar);
