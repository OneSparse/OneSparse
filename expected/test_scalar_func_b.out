set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select '0'::bool::scalar;
 scalar 
--------
 b:f
(1 row)

select '0'::bool::scalar;
 scalar 
--------
 b:f
(1 row)

select '1'::bool::scalar;
 scalar 
--------
 b:t
(1 row)

select set('1'::bool::scalar, 2);
 set 
-----
 b:t
(1 row)

select scalar_bool((0)::bool);
 scalar_bool 
-------------
 b:f
(1 row)

select scalar_bool((0)::bool);
 scalar_bool 
-------------
 b:f
(1 row)

select scalar_bool((1)::bool);
 scalar_bool 
-------------
 b:t
(1 row)

select bool_scalar((0)::bool::scalar);
 bool_scalar 
-------------
 f
(1 row)

select bool_scalar((0)::bool::scalar);
 bool_scalar 
-------------
 f
(1 row)

select bool_scalar((1)::bool::scalar);
 bool_scalar 
-------------
 t
(1 row)

